packages/flutter_test/lib/src/binding.dart

```diff
@@ -709,8 +731,14 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
     final TestExceptionReporter reportTestExceptionBeforeTest = reportTestException;
     final ErrorWidgetBuilder errorWidgetBuilderBeforeTest = ErrorWidget.builder;
 
-    // run the test
+    // run the test including the setup and teardown code that was requested
+    while (_setups.isNotEmpty) {
+      await _setups.removeAt(0)();
+    }
     await testBody();
```

Comment:
I am thinking should we worry about some state reset inside the testBody? https://github.com/flutter/flutter/blob/42c4c30af9a682aaccb3fc4da6e65350fa4adeaf/packages/flutter_test/lib/src/widget_tester.dart#L140

For example if you change the testTextInput in the setup, it will gets in the testBody

Reply:
Good catch. Can we move that + debugResetSemanticsIdCounter(); outside of `binding.runTest`?

Reply:
Another option is to move the setup/teardown execution to inside testBody. I actually like that better.

Reply:
or we can move setups and teardowns inside, i think either way is fine